core/src/main/java/org/pharmgkb/parsers/model/GeneralizedBigDecimal.java

```diff
@@ -0,0 +1,178 @@
+package org.pharmgkb.parsers.model;
+
+import com.google.common.collect.Range;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * A {@link BigDecimal} that can be infinite or NaN.
+ * @author Douglas Myers-Turnbull
+ */
+public class GeneralizedBigDecimal implements Comparable<GeneralizedBigDecimal>, Serializable {
+
+	public static final GeneralizedBigDecimal NAN = new GeneralizedBigDecimal("NaN");
+	public static final GeneralizedBigDecimal POSITIVE_INFINITY = new GeneralizedBigDecimal("Inf");
+	public static final GeneralizedBigDecimal NEGATIVE_INFINITY = new GeneralizedBigDecimal("-Inf");
+	private static final long serialVersionUID = 8608688553173129002L;
+
+	private final Optional<BigDecimal> m_digits;
+	private final String m_string;
```

Comment:
I'm really not a fan of a variable called "string".
Even `m_stringValue` would be better.